### ISSUE TITLE
[PGVector] The scores returned by 'similarity_search_with_score' are NOT proportional to the similarity

### ISSUE THREAD

**Issue by huantt** [langchain-ai/langchain#13437]
### System Info

langchain==0.0.266

### Who can help?

@hwchase17 
@eyurtsev 

### Information

- [X] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

## Behavior
When I call the `vector_store.similarity_search_with_score` function:
- Expected: The returned scores will be proportional to the similarity. This means the higher score, the higher similarity.
- Actual: The scores are proportional to the the distance.

### Problem
- When I call `as_retriever` function with the `score_threshold`, the behavior is wrong. Because when `score_threshold` is declared, it will filter documents that have score greater than or equal to `score_threshold` value. So the top documents that found from pgvector will be filter out while it's the most similar in fact.

### Expected behavior

The returned scores from PGVector queries are proportional to the similarity.
In other words, the higher score, the higher similarity.

**Comment by huantt** [comment]
Related issue:
https://github.com/langchain-ai/langchain/issues/2845

**Comment by huantt** [comment]
Oh, I think that I've understood the folow of PGVector:

Before describe I want to clarify some concepts in Document:
- Distance: higher distance, more least similar
- Score: higher score, more similar

### How PGVector in Langchain works
- In the Database layer: They does not return the score but the distance instead
- In the langchain library layer: They convert the `distance` to the `similarity score` by `distance_strategy`.  When declare PGVector instance, by default, `distance_strategy` is `DistanceStrategy.COSINE`, let's see this function:

```python
    @staticmethod
    def _cosine_relevance_score_fn(distance: float) -> float:
        """Normalize the distance to a score on a scale [0, 1]."""

        return 1.0 - distance
```

It convert the distance to the score.


## So what's wrong?

The one that makes me confuse and should be improved is the `score` concept in `pgvector.py`.
In some functions like `similarity_search_with_score`, the returned value is `distance`, not `score`.

It should be renamed to `similarity_search_with_distinace`
